sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlobTestBase.java

```diff
@@ -266,6 +266,14 @@ protected String setupBlobMatchCondition(BlobAsyncClientBase bac, String match)
         }
     }
 
+    protected Mono<String> setupBlobMatchConditionAsync(BlobAsyncClientBase bac, String match) {
+        if (Objects.equals(match, RECEIVED_ETAG)) {
+            return Objects.requireNonNull(bac.getProperties().map(BlobProperties::getETag));
+        } else {
+            return Mono.just(match == null ? "null" : match);
```

Comment:
I'm not sure if it makes a difference - the reason we are setting it to a null string is to do this check -

`
               String newMatch = tuple.getT2(); //result of setupBlobMatchConditionAsync
                if ("null".equals(newMatch)) {
                    newMatch = null;
                }
`

- As we later pass this value into a new BlobRequestConditions. We tried to return just null instead of the string value, but Mono.just() requires a non-null value.